src/app/shop/catalog/catalog.component.html

```diff
@@ -0,0 +1,41 @@
+<h2>Super Shop!</h2>
+
+<app-item-form></app-item-form>
+<h3>select currency</h3>
+<select [(ngModel)]="currency">
+  <option value="usd">USD</option>
+  <option value="eur">EUR</option>
+  <option value="gbp">GBP</option>
+</select>
+
+
+<h3>
+  <input type="checkbox" [(ngModel)]="onlyDiscounts" (change)="onFilterChange()" />
+  Only discounts
+</h3>
+
+<h2>items</h2>
+
+<app-item-filter></app-item-filter>
+
+<div *ngFor="let item of items| pagination: currentPage: itemsPerPage: this.filterItems">
```

Comment:
En los HTML no es buena práctica usar "this", con que pongas filterItems te debe funcionar.